public/app/features/alerting/unified/api/prometheus.ts

```diff
@@ -30,3 +30,17 @@ export async function fetchRules(dataSourceName: string): Promise<RuleNamespace[
 
   return Object.values(nsMap);
 }
+
+export async function fetchAllRules(): Promise<RuleNamespace[]> {
+  const namespaces = [] as Array<Promise<RuleNamespace[]>>;
+  getAllRulesSourceNames().forEach(async (name) => {
```

Comment:
This has the problem that it won't finish until all of the datasources respond. If ther's a slow one, it will hold up the entire thing.
Maybe make it an action in the same vein as `fetchAllPromAndRulerRulesAction` and then read result form redux store?
That way it will rerender whenever new data comes in and not be impacted by some slow datasource + easy to get the errors

Reply:
I've definitely noticed that certain datasources can slow it down quite a bit so the action may make more sense in this case.